src/Microsoft.DocAsCode.Build.OverwriteDocuments/OverwriteDocumentModelCreater.cs

```diff
@@ -3,13 +3,198 @@
 
 namespace Microsoft.DocAsCode.Build.OverwriteDocuments
 {
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+
+    using Markdig.Syntax;
+
+    using Microsoft.DocAsCode.Common;
     using Microsoft.DocAsCode.Build.Common;
 
     public class OverwriteDocumentModelCreater
     {
-        public OverwriteDocumentModel Create(MarkdownFragmentModel model)
+        public static OverwriteDocumentModel Create(MarkdownFragmentModel model)
+        {
+            var yamlCocdeBlockMetadata = ConvertYamlCodeBlock(model.YamlCodeBlock, model.YamlCodeBlockSource);
+            var contentsMetadata = ConvertContents(model.Contents);
+            return new OverwriteDocumentModel
+            {
+                Uid = model.Uid,
+                Metadata = yamlCocdeBlockMetadata.Concat(contentsMetadata).GroupBy(p => p.Key)
+                    .ToDictionary(g => g.Key, g => g.Last().Value)
+            };
+        }
+
+        public static Dictionary<string, object> ConvertYamlCodeBlock(string yamlCodeBlock, Block yamlCodeBlockSource)
```

Comment:
>public [](start = 8, length = 6)

anywhere else this method is used? Can it be private?

Reply:
Can be internal if only UT needs it

---
In reply to: [163131147](https://github.com/dotnet/docfx/pull/2368#discussion_r163131147) [](ancestors = 163131147)